sensor_control/vision/symbols/src/FrameProc.cpp

```diff
@@ -0,0 +1,85 @@
+#include "FrameProc.h"
+
+
+const int FrameProc::blur_kernel_size = 3;
+const int FrameProc::erode_kernel_size = 3;
+const int FrameProc::dilate_kernel_size = 3;
+
+FrameProc::FrameProc()
+{
+
+	erode_element = getStructuringElement(MORPH_RECT,Size(2*erode_kernel_size + 1,2*erode_kernel_size+1), Point(erode_kernel_size,erode_kernel_size)) ;
+
+	dilate_element = getStructuringElement(MORPH_RECT,Size( 2* dilate_kernel_size + 1, 2* dilate_kernel_size+1 ), Point(dilate_kernel_size, dilate_kernel_size));
+	
+	//OpenCV does 0-180 instead of 0-360, so just divide by 2
+	red = ColorThresh{Scalar(0, 10, 100), Scalar(30, 255, 255)}; //0 to 30/2 (includes orange)
+	red2 = ColorThresh{Scalar(155, 10, 100), Scalar(180, 255, 255)}; // 330/2 to 180
+	blue = ColorThresh{Scalar(90,100,100),Scalar(150,255,255)}; // 210/2 to 270/2
+	green = ColorThresh{Scalar(30, 85, 25),Scalar(85,255,255)}; // 70/2 to 170/2
```

Comment:
Lots of magic constants, you should get these from a config file using ros::param.